Core/src/org/sleuthkit/autopsy/rejview/HexView.java

```diff
@@ -0,0 +1,304 @@
+            hex[2] = ' ';
+            hexSB.append(hex);
+
+            if (b >= ' ' && b <= '~') {
+                asciiSB.append((char) b);
+            } else {
+                asciiSB.append('.');
+            }
+
+            if (i % this.bytesPerLine == this.bytesPerLine - 1) {
+                hexSB.append('\n');
+                asciiSB.append('\n');
+            }
+        }
+
+        offsetView.setText(offsetSB.toString());
+        this.hexViewTextArea.setText(hexSB.toString());
+        this.asciiViewTextArea.setText(asciiSB.toString());
+        this.hexViewTextArea.addCaretListener(hexViewListener);
+        this.asciiViewTextArea.addCaretListener(hexViewListener);
+        this.asciiViewTextArea.setSelectedTextColor(this.asciiViewTextArea.getForeground());
+        this.hexViewTextArea.setSelectedTextColor(this.asciiViewTextArea.getForeground());
+        this.highlighterPainter = new DefaultHighlighter.DefaultHighlightPainter(this.hexViewTextArea.getSelectionColor());
+    }
+
+    /**
+     * Private listener class to listen make changes based on events in the
+     * asciiViewTextArea component and the hexViewTextArea
+     */
+    private class HexViewListener implements CaretListener {
+
+        @Override
+        public void caretUpdate(CaretEvent e) {
+            if (e.getMark() == e.getDot()) {
+                this.clearHighlight();
+            }
+
+            if (e.getSource() == asciiViewTextArea) {
+                int startByte = e.getMark();
+                int endByte = e.getDot();
+
+                if (startByte > endByte) {
+                    int t = endByte;
+                    endByte = startByte;
+                    startByte = t;
+                }
+
+                // the number of line endings before the start,end points
+                int startRows = (startByte - (startByte % bytesPerLine)) / bytesPerLine;
+                int endRows = (endByte - (endByte % bytesPerLine)) / bytesPerLine;
+
+                // the byte index of the start,end points in the ASCII view
+                startByte -= startRows;
+                endByte -= endRows;
+
+                // avoid the loop
+                if (asciiLastSelectionStart == startByte && asciiLastSelectionEnd == endByte) {
+                    return;
+                }
+                asciiLastSelectionStart = startByte;
+                asciiLastSelectionEnd = endByte;
+
+                this.setSelection(startByte, endByte);
+            } else if (e.getSource() == hexViewTextArea) {
+                int startByte = e.getMark();
+                int endByte = e.getDot();
+
+                if (startByte > endByte) {
+                    int t = endByte;
+                    endByte = startByte;
+                    startByte = t;
+                }
+
+                // the number of line endings before the start,end points
+                int startRows = (startByte - (startByte % bytesPerLine)) / (3 * bytesPerLine);
```

Comment:
We should probably make the "3" into a constant, since we are using it in multiple places.

Reply:
Will do.